### DISCUSSION TITLE
What is the correct syntax for including SQL values AND rowsAsArray option in a query?

### DISCUSSION THREAD

**Discussion by millerbryan** [sidorares/node-mysql2#3095] [Q&A]
I've looked at all the examples in the docs which use `rowsAsArray` and it isn't clear how to specify it in a query (rather than in the connection) when passing in SQL values.  Is this possible to pass both?

```
// simplified code
const sql = "select county_name from static_web_data.state_counties where state_name = ?";
const state = "TX"

        try {
          const [rows, fields] = await pool.query({sql, [state], rowsAsArray: true});
          if (rows) {
            return rows;
          }
        } catch (err) {
          console.log(err);
        }
```

**Comment by wellwelwel** [comment]
Hi, @millerbryan 🙋🏻‍♂️

You can set it in both connection and query levels:

https://github.com/sidorares/node-mysql2/blob/d02aa7c9a0cd6d91a3df25d768420cf83b044bbd/website/docs/index.mdx#L452-L457

https://github.com/sidorares/node-mysql2/blob/d02aa7c9a0cd6d91a3df25d768420cf83b044bbd/website/docs/index.mdx#L481-L491

You can see the [docs#array-results](https://sidorares.github.io/node-mysql2/docs#array-results) section for more details.

**Comment by millerbryan** [comment]
Thank you @wellwelwel  - I saw those examples in the docs.

However, my question is how to set rowsAsArray in the query when I am _also_ passing in SQL values to the query?

Please see my code snippet in the OP.

Is there a keyword for SQL values like this for SQL values?

```
try { 
   const [results, fields] = await conn.query({ 
     sql: 'SELECT 1 AS `foo`, 2 AS `foo`', 
     values: [state],
     rowsAsArray: true, 
   }); 
```
